### ISSUE TITLE
Installing the portal using helm

### ISSUE THREAD

**Issue by MarkAshraf96** [eclipse-tractusx/portal-cd#9]
Hello,

I am currently trying to install the portal on my machine using that command
`helm install portal tractusx-dev/portal --timeout 10m0s`
and I face this error,
`Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition`

You can see that I already raised the timeout from the default 5mins to 10mins and still I get that error.

Any Advice?

Thank You in advance

**Comment by evegufy** [comment]
Sure, thanks for the feedback!
My first guess would be that you're running out of resources. But I can't reproduce the behavior.
Where are you attempting to install it to? On a minikube?
I assume that you're attempting to install the latest version available on tractusx-dev (1.2.1 chart version / 1.2.0 app version), correct?

**Comment by MarkAshraf96** [comment]
I am installing it on a Minikube yes,
I followed the steps:

- Installed the requirement
`helm repo add bitnami https://charts.bitnami.com/bitnami
helm install my-release bitnami/postgresql`

- Added the repo
` helm repo add tractusx-dev https://eclipse-tractusx.github.io/charts/dev`

- Installed the repo
`helm install portal tractusx-dev/portal`

Also another teammate of mine had installed it sucessfully last week, when tries now gets the same error `failed post-install: timed out waiting for the condition`

![image](https://user-images.githubusercontent.com/40201647/231755016-9e7cc0dd-ebb0-489f-adcd-0ec4179459a6.png)

**Comment by MarkAshraf96** [comment]
I have tried with the second approach but still it didn't work. And gave the same error.

```Console
markashraf@iot-owner-8d46dd85 CatenaX % minikube delete                                           
🔥  Deleting "minikube" in qemu2 ...
💀  Removed all traces of the "minikube" cluster.
markashraf@iot-owner-8d46dd85 CatenaX % minikube start                                            
😄  minikube v1.30.1 on Darwin 13.2 (arm64)
✨  Automatically selected the qemu2 driver
🌐  Automatically selected the builtin network
❗  You are using the QEMU driver without a dedicated network, which doesn't support `minikube service` & `minikube tunnel` commands.
To try the dedicated network see: https://minikube.sigs.k8s.io/docs/drivers/qemu/#networking
👍  Starting control plane node minikube in cluster minikube
🔥  Creating qemu2 VM (CPUs=2, Memory=4000MB, Disk=20000MB) ...
🐳  Preparing Kubernetes v1.26.3 on Docker 20.10.23 ...
    ▪ Generating certificates and keys ...
    ▪ Booting up control plane ...
    ▪ Configuring RBAC rules ...
🔗  Configuring bridge CNI (Container Networking Interface) ...
    ▪ Using image gcr.io/k8s-minikube/storage-provisioner:v5
🔎  Verifying Kubernetes components...
🌟  Enabled addons: storage-provisioner, default-storageclass
🏄  Done! kubectl is now configured to use "minikube" cluster and "default" namespace by default
markashraf@iot-owner-8d46dd85 CatenaX % kubectl get pods
No resources found in default namespace.
markashraf@iot-owner-8d46dd85 CatenaX % kubectl get service                                       
NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   20s
markashraf@iot-owner-8d46dd85 CatenaX % helm install portal -f values-own.yaml tractusx-dev/portal
Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition
markashraf@iot-owner-8d46dd85 CatenaX % kubectl get pods                                          
NAME                                       READY   STATUS             RESTARTS       AGE
administration-service-69d4b6bd85-72h45    0/1     CrashLoopBackOff   5 (82s ago)    5m50s
administration-service-69d4b6bd85-c9js4    0/1     CrashLoopBackOff   5 (74s ago)    5m50s
administration-service-69d4b6bd85-w7qh4    0/1     CrashLoopBackOff   5 (77s ago)    5m50s
assets-76d57d5c7f-kvzjn                    0/1     CrashLoopBackOff   5 (94s ago)    5m50s
assets-76d57d5c7f-nhp5g                    0/1     CrashLoopBackOff   5 (92s ago)    5m50s
assets-76d57d5c7f-v2wjt                    0/1     CrashLoopBackOff   5 (77s ago)    5m50s
checklist-worker-28030170-ds7jt            0/1     CrashLoopBackOff   5 (45s ago)    3m57s
marketplace-app-service-7cd6957f97-cd6sm   0/1     CrashLoopBackOff   5 (68s ago)    5m50s
marketplace-app-service-7cd6957f97-th6s7   0/1     CrashLoopBackOff   5 (81s ago)    5m49s
marketplace-app-service-7cd6957f97-v5fdn   0/1     CrashLoopBackOff   5 (68s ago)    5m50s
notification-service-786c767bb8-6g76j      0/1     CrashLoopBackOff   5 (88s ago)    5m50s
notification-service-786c767bb8-8bhkz      0/1     CrashLoopBackOff   5 (84s ago)    5m50s
notification-service-786c767bb8-cmt6l      0/1     CrashLoopBackOff   5 (67s ago)    5m50s
portal-59d56858f7-chcpn                    0/1     CrashLoopBackOff   5 (79s ago)    5m50s
portal-59d56858f7-dc26v                    0/1     CrashLoopBackOff   5 (95s ago)    5m50s
portal-59d56858f7-hqdsb                    0/1     CrashLoopBackOff   5 (76s ago)    5m50s
portal-backend-postgresql-primary-0        0/1     CrashLoopBackOff   5 (115s ago)   5m50s
portal-backend-postgresql-read-0           0/1     CrashLoopBackOff   5 (114s ago)   5m50s
portal-migrations-5gs6f                    0/1     Error              0              4m9s
portal-migrations-8t77q                    0/1     Error              0              2m49s
portal-migrations-mj92m                    0/1     Error              0              5m50s
portal-migrations-np6sh                    0/1     Error              0              4m58s
portal-migrations-v7jms                    0/1     Error              0              3m39s
portal-migrations-xjh2z                    0/1     Error              0              82s
provisioning-service-5cfc9c7766-cpr95      0/1     CrashLoopBackOff   5 (101s ago)   5m50s
provisioning-service-5cfc9c7766-lhj5n      0/1     CrashLoopBackOff   5 (88s ago)    5m50s
provisioning-service-5cfc9c7766-qzkxx      0/1     CrashLoopBackOff   5 (86s ago)    5m50s
registration-6db87f8d8b-bfldb              0/1     CrashLoopBackOff   5 (83s ago)    5m50s
registration-6db87f8d8b-t2rlq              0/1     CrashLoopBackOff   5 (79s ago)    5m50s
registration-6db87f8d8b-z4m8n              0/1     CrashLoopBackOff   5 (105s ago)   5m50s
registration-service-6586d974d6-7hgm8      0/1     CrashLoopBackOff   5 (98s ago)    5m50s
registration-service-6586d974d6-bsq6h      0/1     CrashLoopBackOff   5 (85s ago)    5m50s
registration-service-6586d974d6-p5dqz      0/1     CrashLoopBackOff   5 (96s ago)    5m50s
services-service-57c4f7877f-5cf49          0/1     CrashLoopBackOff   5 (87s ago)    5m50s
services-service-57c4f7877f-lml7k          0/1     CrashLoopBackOff   5 (84s ago)    5m50s
services-service-57c4f7877f-qhfnv          0/1     CrashLoopBackOff   5 (80s ago)    5m50s
markashraf@iot-owner-8d46dd85 CatenaX %                 
```

**Comment by MarkAshraf96** [comment]
I had the same error with the installation before, the installation worked on a colleague's laptop.

However, with the new version of the chart 1.3.0-RC2.
It still gets the same error, but I have most of the pods up and running.
Is that the expected output?
```console
markashraf@iot-owner-8d46dd85 CatenaX % helm install portal -f values-own.yaml tractusx-dev/portal
Error: INSTALLATION FAILED: failed post-install: timed out waiting for the condition
markashraf@iot-owner-8d46dd85 CatenaX % kubectl get service                                       
NAME                                   TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
administration-service                 ClusterIP   10.98.210.254    <none>        8080/TCP   7m40s
assets                                 ClusterIP   10.101.142.90    <none>        8080/TCP   7m40s
kubernetes                             ClusterIP   10.96.0.1        <none>        443/TCP    12m
marketplace-app-service                ClusterIP   10.100.72.78     <none>        8080/TCP   7m40s
notification-service                   ClusterIP   10.105.224.5     <none>        8080/TCP   7m40s
portal                                 ClusterIP   10.108.205.174   <none>        8080/TCP   7m40s
portal-backend-postgresql-primary      ClusterIP   10.98.111.122    <none>        5432/TCP   7m40s
portal-backend-postgresql-primary-hl   ClusterIP   None             <none>        5432/TCP   7m40s
portal-backend-postgresql-read         ClusterIP   10.110.212.130   <none>        5432/TCP   7m40s
portal-backend-postgresql-read-hl      ClusterIP   None             <none>        5432/TCP   7m40s
provisioning-service                   ClusterIP   10.105.207.124   <none>        8080/TCP   7m40s
registration                           ClusterIP   10.107.178.93    <none>        8080/TCP   7m40s
registration-service                   ClusterIP   10.102.222.105   <none>        8080/TCP   7m40s
services-service                       ClusterIP   10.97.16.71      <none>        8080/TCP   7m40s
markashraf@iot-owner-8d46dd85 CatenaX % kubectl get pods                                          
NAME                                       READY   STATUS             RESTARTS        AGE
administration-service-69d4b6bd85-mq5sl    0/1     OOMKilled          3 (2m30s ago)   7m53s
administration-service-69d4b6bd85-q48bl    1/1     Running            3 (2m17s ago)   7m53s
administration-service-69d4b6bd85-xz5l4    0/1     OOMKilled          2 (3m8s ago)    7m53s
assets-76d57d5c7f-7r678                    1/1     Running            0               7m53s
assets-76d57d5c7f-cfbm4                    1/1     Running            0               7m53s
assets-76d57d5c7f-pt7st                    1/1     Running            0               7m53s
checklist-worker-28031550-d9hnj            1/1     Running            0               4m40s
marketplace-app-service-7cd6957f97-7q6fw   1/1     Running            3 (111s ago)    7m53s
marketplace-app-service-7cd6957f97-h8dfp   1/1     Running            3 (94s ago)     7m53s
marketplace-app-service-7cd6957f97-n89fw   1/1     Running            3 (86s ago)     7m53s
notification-service-786c767bb8-bx5dz      1/1     Running            3 (64s ago)     7m53s
notification-service-786c767bb8-tk6w5      0/1     CrashLoopBackOff   2 (19s ago)     7m53s
notification-service-786c767bb8-xb7bz      1/1     Running            3 (2m3s ago)    7m53s
portal-59d56858f7-9p4dg                    1/1     Running            0               7m53s
portal-59d56858f7-v6lbx                    1/1     Running            0               7m53s
portal-59d56858f7-vthh6                    1/1     Running            0               7m53s
portal-backend-postgresql-primary-0        1/1     Running            0               7m53s
portal-backend-postgresql-read-0           1/1     Running            0               7m53s
portal-migrations-9vzl7                    0/1     OOMKilled          0               7m53s
portal-migrations-fkd9j                    1/1     Running            0               115s
provisioning-service-5cfc9c7766-8qvgj      1/1     Running            3 (58s ago)     7m53s
provisioning-service-5cfc9c7766-nq4lw      1/1     Running            3 (54s ago)     7m53s
provisioning-service-5cfc9c7766-pkxxw      1/1     Running            2 (2m10s ago)   7m53s
registration-6db87f8d8b-crcm9              1/1     Running            0               7m53s
registration-6db87f8d8b-p9fcr              1/1     Running            3 (86s ago)     7m53s
registration-6db87f8d8b-tmchl              1/1     Running            0               7m53s
registration-service-6586d974d6-7cqg8      0/1     OOMKilled          2 (2m49s ago)   7m53s
registration-service-6586d974d6-vmwhl      1/1     Running            3 (105s ago)    7m53s
registration-service-6586d974d6-zckgn      1/1     Running            3 (77s ago)     7m53s
services-service-57c4f7877f-5c6nq          1/1     Running            3 (2m6s ago)    7m53s
services-service-57c4f7877f-8gmp8          1/1     Running            3 (70s ago)     7m53s
services-service-57c4f7877f-grd6m          1/1     Running            3 (99s ago)     7m53s

```

**Comment by evegufy** [comment]
I think you're running OOM. Can you give to your minikube more resources?
I'm start minikube with default settings, and can't reproduce it.
Another option, you could try, is managing resources in your own values file:
https://github.com/eclipse-tractusx/portal-cd/blob/main/charts/portal/values.yaml#L590

**Comment by MarkAshraf96** [comment]
I have tried both approaches:

- I modified the resources assigned and limits in my own values file to memory: 4000Mi. I get a positive feedback for the helm install command, but most of the pods kept in the pending status.

- I modified the Minikube assigned resources of memory to 6000MB it gave me the post-install time out waiting for the condition , and there was one pod that had and OOMKilled status.
with 7000MB it also gave me the post-install time out waiting for the condition, all the pods has a running status initially then the checklist-worker gives an error.
